src/plugins/data/server/search/search_service.ts

```diff
@@ -309,25 +312,27 @@ export class SearchService implements Plugin<ISearchSetup, ISearchStart> {
 
   private cancelSessionSearches = async (deps: SearchStrategyDependencies, sessionId: string) => {
     const searchIdMapping = await deps.searchSessionsClient.getSearchIdMapping(sessionId);
-
-    for (const [searchId, strategyName] of searchIdMapping.entries()) {
-      const searchOptions = {
-        sessionId,
-        strategy: strategyName,
-        isStored: true,
-      };
-      this.cancel(deps, searchId, searchOptions);
-    }
+    await Promise.all(
+      Object.keys(searchIdMapping).map((searchId) => {
+        const strategyName = searchIdMapping.get(searchId);
+        const searchOptions = {
+          sessionId,
+          strategy: strategyName,
+          isStored: true,
+        };
+        return this.cancel(deps, searchId, searchOptions);
+      })
+    );
   };
 
   private cancelSession = async (deps: SearchStrategyDependencies, sessionId: string) => {
     const response = await deps.searchSessionsClient.cancel(sessionId);
-    this.cancelSessionSearches(deps, sessionId);
+    await this.cancelSessionSearches(deps, sessionId);
```

Comment:
I was under the impression that we didn't want to fail the call to `cancelSession` if cancelling one of the search requests fails, but isn't that what will happen now with your changes?

Reply:
I think we talked about still changing the state of the object to canceled, even if some searches deletion failed